### PR TITLE
Documentation review

### PR DESCRIPTION
**Pull request by ericof**
* Remove pyramid_autodoc
* Upgrade Sphinx
* Add make all target

### CODE REVIEW

File: docs/source/narrative/contributing/develop.rst [hunk 1]
```diff
@@ -8,40 +8,61 @@ Developing Websauna
...
-* Python 3.5 with development headers (``apt-get install python-dev``)
-* virtualenv (``apt-get install python-virtualenv``)
-* pip (``apt-get install python-pip``)
-* git (``apt-get install git``)
-* PhantomJS headless browser (``apt-get install phantomjs``)
-* Redis (``apt-get install redis``)
+In order to develop Websauna, the following requirements need to be installed on your system:
+
+    * GCC, make, and similar
+    * Python 3.5 with development headers
```
Comment (jsbueno) [inline docs/source/narrative/contributing/develop.rst]: "or above" ?

File: docs/source/reference/glossary.rst [hunk 2]
```diff
@@ -66,12 +63,18 @@ Many glossary descrpitions are taken from `Wikipedia <https://en.wikipedia.org/>
...
         Development is the stage of the deployment when the developer is locally working on the code on a local machine. See :ref:`development.ini`.
 
+    development.ini
+      The default configuration file when you run Websauna on your local computer when doing development. For more information see :doc:`configuration <../reference/config>`.
+
+    deform
+        A form framework suggested by Websauna. :doc:`Read Websauna form documentation <../narrative/form/form>`. `See widget samples <http://demo.substanced.net/deformdemo/>`__. `More info <http://deform.readthedocs.org/en/latest/>`__.
+
     duplicity
         Duplicity backs directories by producing encrypted tar-format volumes and uploading them to a remote or local file server. Because duplicity uses librsync, the incremental archives are space efficient and only record the parts of files that have changed since the last backup. Because duplicity uses GnuPG to encrypt and/or sign these archives, they will be safe from spying and/or modification by the server. `More info <http://duplicity.nongnu.org/>`__.
```
Comment (jsbueno) [inline docs/source/reference/glossary.rst]: This line should be duplicated.
(cheapest ever easter egg)

File: docs/source/narrative/contributing/develop.rst (hunk 1 above)
Comment (ericof) [reply docs/source/narrative/contributing/develop.rst]: Not yet. We are still supporting only 3.4 and 3.5 (even though in the Pyramid 1.9 branch I added 3.6 support as well).

File: docs/source/reference/glossary.rst (hunk 2 above)
Comment (ericof) [reply docs/source/reference/glossary.rst]: Would be awesome but I think Sphinx will yell at me :-D

**Pull request merged**